brain_node:
  ros__parameters:
    game:
      team_id: 29 # must be consistant with GameControl
      player_id: 1 # 1 | 2 | 3 | 4 | 5
      field_type: "adult_size" # adult_size | kid_size | robo_league
      player_role: "striker" # striker | goal_keeper
      treat_person_as_robot: false 
      number_of_players: 3
      
    robot:
      robot_height: 0.9 
      odom_factor: 1.2 
      vx_factor: 0.50 
      yaw_offset: 0.0 
      vx_limit: 1.0 
      vy_limit: 0.4
      vtheta_limit: 1.2 

    strategy:
      ball_confidence_threshold: 50.0 
      ball_memory_timeout: 5.0 
      tm_ball_dist_threshold: 2.0 
      limit_near_ball_speed: false 
      near_ball_speed_limit: 0.2 
      near_ball_range: 3.0
      abort_kick_when_ball_moved: true 
      cooperation: 
        enable_role_switch: false 
        ball_control_cost_threshold: 20.0 

    obstacle_avoidance:
      depth_sample_step: 16 
      obstacle_min_height: 0.45 
      grid_size: 0.2 
      max_x: 3.0 
      max_y: 5.0 
      exclusion_x: 0.4 
      exclusion_y: 0.5
      ball_exclusion_radius: 0.0 
      ball_exclusion_height: 0.3 
      occupancy_threshold: 2000 
      collision_threshold: 0.3 
      safe_distance: 2.0 
      avoid_secs: 3.0 
      enable_freekick_avoid: false 
      freekick_start_placing_safe_distance: 0.1 
      freekick_start_placing_avoid_secs: 2.5 
      obstacle_memory_msecs: 500. 
      avoid_during_kick: false 
      kick_ao_safe_dist: 3.0 
      kick_ao_use_shoot: false 
      avoid_during_chase: true 
      chase_ao_safe_dist: 3.5 

    locator:
      min_marker_count: 5
      max_residual: 0.35 

    enable_com: false 

    rerunLog:
      enable_tcp: false 
      server_ip: "192.168.10.110:9876" 
      enable_file: true 
      log_dir: "/home/booster/Workspace/rrlog"
      max_log_file_mins: 5.0 
      img_interval: 1 
    
    recovery:
      retry_max_count: 2

    vision:
        # image_topic: "/camera/camera/color/image_raw" #realsense
        # depth_image_topic: "/camera/camera/aligned_depth_to_color/image_raw" #realsense
        # image_topic: "/zed/zed_node/left/image_rect_color" #zed
        # depth_image_topic: "/zed/zed_node/depth/depth_registered" #zed
        # image_topic: "/camera/color/image_raw" # orbbec
        # depth_image_topic: "/camera/depth/image_raw" # orbbec
        image_topic: "/booster_camera_bridge/StereoNetNode/rectified_image" # d-robotics
        depth_image_topic: "/booster_camera_bridge/StereoNetNode/stereonet_depth" # d-robotics
        cam_pixel_width: 544. # 相机像素宽度, d-robotics
        cam_pixel_height: 448. # 相机像素高度, d-robotics
        # cam_pixel_width: 1280. # 相机像素宽度, realsense orbbec 与 zed 相同
        # cam_pixel_height: 720. # 相机像素高度, realsense orbbec 与 zed 相同
        # cam_fov_x: 90.0 # 相机视场角, 单位: 度, realsense 和 orbbec相同
        # cam_fov_y: 60.0 # 相机视场角, 单位: 度, realsense 和 orbbec相同
        # cam_fov_x: 110.0 # 相机视场角, 单位: 度, zed
        # cam_fov_y: 70.0 # 相机视场角, 单位: 度, zed
        cam_fov_x: 69.4 # 相机视场角, 单位: 度, d-robotics
        cam_fov_y: 42.5 # 相机视场角, 单位: 度, d-robotics
        # TODO 相机参数以及点云处理, 未来应该转移到 vision 模块中. 或使 vision 模块发送 TF 消息.
    
    game_control_ip: "127.0.0.1"

    # for fun and debug
    sound:
      enable: false 
      sound_pack: "espeak" 